contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java

```diff
@@ -92,9 +92,16 @@ public void bind(ManagedScanFramework framework) { }
     @Override
     public ManagedReader<SchemaNegotiator> next() {
 
+      // Get the expected input type
+      String inputType = subScan.tableSpec().connectionConfig().inputType();
+
       // Only a single scan (in a single thread)
       if (count++ == 0) {
-        return new HttpBatchReader(subScan);
+        if (inputType.equalsIgnoreCase("csv")) {
+          return new HttpCSVBatchReader(subScan);
+        } else {
+          return new HttpBatchReader(subScan);
+        }
```

Comment:
Github won't let me leave the comment for `else` block down below, but that `return null;` can go as default (without `else`).

Reply:
Fixed